diff --git a/ckanext/datapusher/logic/action.py b/ckanext/datapusher/logic/action.py
index 5a5f51967..a64313e04 100644
--- a/ckanext/datapusher/logic/action.py
+++ b/ckanext/datapusher/logic/action.py
@@ -301,7 +301,7 @@ def datapusher_status(context, data_dict):
                         log['timestamp'], "%Y-%m-%dT%H:%M:%S.%f")
                     date = datetime.datetime.utcfromtimestamp(
                         time.mktime(date))
-                    log['timestamp'] = date
+                    log['timestamp'] = date.isoformat()
         except (requests.exceptions.ConnectionError,
                 requests.exceptions.HTTPError):
             job_detail = {'error': 'cannot connect to datapusher'}
