diff --git a/ckanext/reclineview/plugin.py b/ckanext/reclineview/plugin.py
index debcc574e..b159a7d6c 100644
--- a/ckanext/reclineview/plugin.py
+++ b/ckanext/reclineview/plugin.py
@@ -152,7 +152,7 @@ class ReclineGraphView(ReclineViewBase):
 
     datastore_fields = []
 
-    datastore_field_types = ['numeric', 'int4', 'timestamp']
+    datastore_field_types = ['numeric', 'int4', 'timestamp', 'text']
 
     def list_graph_types(self):
         return [t['value'] for t in self.graph_types]
